src/components/modals/profile/library/BaseProfileLibraryFolderImportModal/ImportSection.vue

```diff
@@ -87,35 +87,20 @@ export default {
       )
     },
     formatFile (file) {
-      const reader = new jsmediatags.Reader(
-        file
-      )
-
-      const handleSuccess = ({ tags }) => {
-        if (this.isMounted) {
-          this.addToSuccessFiles(
-            { tags, file }
-          )
-        }
-      }
-
-      const handleError = () => {
-        if (this.isMounted) {
-          this.addToErrorFiles(
-            { file }
-          )
-        }
-      }
-
-      reader.read({
-        onSuccess: handleSuccess,
-        onError: handleError
+      musicMetadata.parseFile(file.path).then(metadata => {
```

Comment:
@Borewit Please bring back `handleSuccess/handleError` functions and place that login in there.

Reply:
music-metadata returns a promise. There are clear patters how to handle fulfilled (success) and rejected (error) promises.

I am all for verbosity, but wrapping function into function without any reasons defeats my sense of logic.

Maybe you should rethink that pattern. Also assigning anonymous functions to const rather then just using named functions is my opinion only over-complicating things.